File: solutions/java/lasagna/1/src/main/java/Lasagna.java
public class Lasagna {
    public int expectedMinutesInOven() {
        return 40;
    }

    public int remainingMinutesInOven(int elapsedMinutes) {
        return expectedMinutesInOven() - elapsedMinutes;
    }

    public int preparationTimeInMinutes(int numberOfLayers) {
        return numberOfLayers * 2;
    }

    public int totalTimeInMinutes(int numberOfLayers, int elapsedMinutes) {
        return preparationTimeInMinutes(numberOfLayers) + elapsedMinutes;
    }
}

File: solutions/swift/lasagna/1/Sources/Lasagna/Lasagna.swift
let expectedMinutesInOven = 40

func remainingMinutesInOven(elapsedMinutes: Int) -> Int {
  return expectedMinutesInOven - elapsedMinutes
}

func preparationTimeInMinutes(layers numberOfLayers: Int) -> Int {
  return numberOfLayers * 2
}

func totalTimeInMinutes(layers numberOfLayers: Int, elapsedMinutes: Int) -> Int {
  return preparationTimeInMinutes(layers: numberOfLayers) + elapsedMinutes
}
